File: Nov2020/30Nov/EasyOne.java
/*
Problem Name: Easy one
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/easy-one-8/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        HashMap<Long,Integer> map = new HashMap<Long,Integer>();
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt(), q = in.nextInt();
            for(;n>0;n--){
                long ele = in.nextLong();
                if(map.containsKey(ele))
                    map.put(ele,map.get(ele) + 1);
                else
                    map.put(ele,1);
            }
            for(;q>0;q--){
                long ele = in.nextLong();
                if(map.containsKey(ele))
                    out.println(map.get(ele));
                else
                    out.println(0);
            }
            map.clear();
        }
        out.flush();
    }
}

File: Nov2020/30Nov/EasyOne.cpp
/*
Problem Name: Easy one
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/easy-one-8/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    test(t){
        int n,q,val;
        cin>>n>>q;
        map<int,int> m;
        for(int i=0;i<n;++i){
            cin>>val;
            m[val]++;
        }
        for(int i=0;i<q;++i){
            cin>>val;
            cout<<m[val]<<"\n";
        }
    }
    return 0;
}
